go/bundle/encoder.go

```diff
@@ -95,9 +95,12 @@ type indexSection struct {
 	bytes []byte
 }
 
-func (is *indexSection) addExchange(e *Exchange, offset, length int) error {
+func (is *indexSection) addExchange(e *Exchange, offset, length int, ver version.Version) error {
 	variants := normalizeHeaderValues(e.Response.Header[http.CanonicalHeaderKey("variants")])
 	variantKey := normalizeHeaderValues(e.Response.Header[http.CanonicalHeaderKey("variant-key")])
+	if (!ver.SupportsVariants() && (variants != "" || variantKey != "")) {
+		panic("Variants included in the exchange, but the specific bundle version '" + ver + "' does not support it.")
```

Comment:
I think `b2` bundle should be "transparent" regarding the `variants` / `variant-key` headers, i.e. these headers are just stored in the bundle without any special treatment.